packages/amazon-sumerian-hosts-core/src/core/awspack/LexFeature.js

```diff
@@ -0,0 +1,287 @@
+   */
+  processWithAudio(inputAudio, sourceSampleRate, config = {}) {
+    const audio = this._prepareAudio(inputAudio, sourceSampleRate);
+    return this._process('audio/x-l16; rate=16000', audio, config);
+  }
+
+  /**
+   * Sends text user input to Amazon Lex.
+   *
+   * @param {String} inputText - Text to send to lex bot
+   * @param {Object=} config - Optional config for overriding lex bot info
+   * @param {string=} config.botName - The name of the lex bot.
+   * @param {string=} config.botAlias - The alias of the lex bot.
+   * @param {string=} config.userId - The userId used to keep track of the session with lex bot. 
+   * 
+   * @returns {Deferred} A Promise-like object that resolves to a Lex response object. 
+   * For details on the structure of that response object see: https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/LexRuntime.html#postContent-property
+   */
+  processWithText(inputText, config = {}) {
+    return this._process('text/plain; charset=utf-8', inputText, config);
+  }
+
+  _process(contentType, inputStream, config) {
+    const settings = this._validateConfig(config);
+    const lexSettings = {
+      botName: settings.botName,
+      botAlias: settings.botAlias,
+      contentType,
+      inputStream,
+      userId: settings.userId
+    };
+    return new Deferred((resolve, reject) => {
+      this.constructor.SERVICES.lexRuntime.postContent(lexSettings, (error, data) => {
+        if (error) {
+          return reject(error);
+        }
+        return resolve(data);
+      });
+    }).then(data => {
+      if (!data.message) {
+        if (data.encodedMessage) {
+          data.message = atob(data.encodedMessage);
+        }
+        else {
+          data.message = '';
+        }
+      }
+
+      if (this._messenger && this._messenger.emit) {
+        this._messenger.emit(this.constructor.EVENTS.lexResponse, data);
+      }
+
+      return data;
+    });
+  }
+
+  _validateConfig(config) {
+    let settings = {};
+
+    settings.botName = config.botName ? config.botName : this._botName;
+    settings.botAlias = config.botAlias ? config.botAlias : this._botAlias;
+    settings.userId = config.userId ? config.userId : this._userId;
+
+    if (
+      settings.botName == undefined ||
+      settings.botAlias == undefined ||
+      settings.userId == undefined
+    )
+    {
+      throw new Error(
+        'Cannot process lex request. All arguments must be defined.'
+      );
+    }
+
+    return settings;
+  }
+
+  _prepareAudio(audioBuffer, sourceSampleRate) {
+    const downsampledAudio = LexUtils.downsampleAudio(audioBuffer, sourceSampleRate, this.constructor.LEX_DEFAULTS.SampleRate);
+    const encodedAudio = LexUtils.encodeWAV(downsampledAudio, this.constructor.LEX_DEFAULTS.SampleRate);
+
+    return new Blob([encodedAudio], {type: 'application/octet-stream'});
```

Comment:
is this the best mime type for this audio?

Reply:
I don't think this matters since the actually content-type is specified within postContent param.